plumber/asset/model.py

```diff
@@ -94,9 +95,25 @@ def import_model(self, model: Model, collection: Collection) -> None:
                 # this only gets called if there is 1 mesh
                 parent_obj = mesh_obj
 
-        self.imported_objects[original_name.lower()] = ModelState(
-            parent_obj, children, collection
-        )
+        model_state = ModelState(parent_obj, children, collection)
+        self.imported_objects[original_name.lower()] = model_state
+
+        # Track this model for batch operations
+        self.batch_imported.append(parent_obj)
```

Comment:
Add boolean which is checks if this tracking is needed, which is only true when the import is an MDL import. If it is false, do not append for performance